src/snowflake/snowpark/session.py

```diff
@@ -716,6 +717,7 @@ def get_packages(self) -> Dict[str, str]:
     def add_packages(
         self,
         *packages: Union[str, ModuleType, Iterable[Union[str, ModuleType]]],
+        upload_custom_packages: bool = False,
         force_push: bool = True,
```

Comment:
```suggestion
        *packages: Union[str, ModuleType, Iterable[Union[str, ModuleType]]],
        *,
        upload_custom_packages: bool = False,
        force_push: bool = True,
```

Reply:
I would also think that using how we enable or disable sql simplifier should also be how to enable or disable this feature. @sfc-gh-yixie what do you think?

Reply:
> 
The extra star is not needed here because packages argument is specified as `*packages`.
For example, `session.add_packages("sktime", True, True)` would result in errors about "True" not being a valid package.

Reply:
I think that's the intention of adding `*`. We want users to specify packages freely but if they want a force upload they need to do `session.add_packages(package1, package2, force_push=True)`

Reply:
Quick clarification; `*packages,` and `*,` are equivalent. Should I add a test case to ensure that experimental parameters cannot be specified accidentally?

Reply:
they are not equivalent but it seems they can't be added together: https://peps.python.org/pep-3102/

Reply:
@sfc-gh-aalam That sounds a good way to set the parameter on the session because it will impact multiple functions (session.add_requirements, session.udf.register, etc) with a single config. 
Sophie added `session.conf` to add configs at the client. We can use it. 
@sfc-gh-vnayak Then I guess you also need to move `force_push` to the session level?

Reply:
Switched to parameter.

`force_push` felt too specific to move to session level, instead I have limited usage of `force_push` via `Session` functions only (like `add_requirements` or `add_packages`) and set it to default to `False`.